docs/zkapps/o1js/bitwise-gadgets.mdx

```diff
@@ -0,0 +1,420 @@
+---
+title: Bitwise Gadgets
+hide_title: true
+description: Bitwise gadgets in o1js.
+keywords:
+  - gadgets
+  - bitwise
+  - o1js
+---
+
+:::info
+
+Please note that zkApp programmability is not yet available on Mina Mainnet, but zkApps can now be deployed to Berkeley Testnet.
+
+:::
+
+# Bitwise Gadgets in o1js
+
+Bitwise operations work on 2-bit patterns of equal lengths by positionally matching their individual bits. 
+
+- [and](#and)
+- [compactMultiRangeCheck](#compactmultirangecheck)
+- [leftShift](#leftshift)
+- [multiRangeCheck](#multirangecheck)
+- [not](#not)
+- [rangeCheck64](#rangecheck64)
+- [rightShift](#rightshift)
+- [rotate](#rotate)
+- [xor](#xor)
+
+## and
+
+```ts
+and(a: Field, b: Field, length: number) {
+    return and(a, b, length);
+```
+
+The bitwise `and()` gadget on Field elements is equivalent to the [bitwise AND (&)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Bitwise_AND) operator in JavaScript. 
```

Comment:
@garwalsh  I have basic questions:
1. how do we want to "write" the name of the gadget?
For example, I see variations on `and()` or `and` or AND
2. is it a gadget or a gate? we use both terms